Revamp docs

michaelkirk opened the pull request:
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Some of it I hope is uncontroversial:

- there were references to old serialization engines.
- linkified more entities

Some of it is potentially more controversial - and I'm open to changing
it based on feedback:

Primarily, I attempted to make the docs more perusable for people that
like to gloss over and find examples, while maintaining a decent amount
of coherent narrative for the blessed people that actually read docs top
to bottom.

michaelkirk commented on src/lib.rs:
```diff
@@ -14,147 +14,140 @@
-//! # #[cfg(feature = "geo-types")]
-//! let polys = vec![poly];
-//! # #[cfg(feature = "geo-types")]
-//! let gc = GeometryCollection::from_iter(polys);
```
ack, I lost this example. I'm going to re-add it. One minute!

michaelkirk replied:
Ok, restored!

I did move it to the `### From geo-types to geojson` section below. It seemed a little out of place as the very first example in the doc.

urschrei commented on src/lib.rs:
````diff
@@ -14,147 +14,140 @@
+//! geojson = "*"
 //! ```
+//!
+//! If you want to use GeoJSON as input to or output from a geometric processing crate like
````
Maybe `geometry` rather than `geometric`?

urschrei commented on src/lib.rs:
```diff
@@ -14,147 +14,140 @@
+//!    field or any other properties you'd like associated with the `Geometry`.
+//! 3. A [`FeatureCollection`] is a list of one or more `Feature`s.
+//!
+//! Because [`Feature`] and [`FeatureCollection`] are more flexible, bare [`Geometry`] GeoJSONs are
```
Very much down to individual style, but I usually see `GeoJSON is` rather than `GeoJSONs` are

urschrei commented on src/lib.rs:
```diff
@@ -14,147 +14,140 @@
+//! `GeometryCollection`](Value::GeometryCollection). Do you want it converted to a single
+//! [`Feature`] whose geometry is a [`GeometryCollection`](Value::GeometryCollection), or do you
+//! want a [`FeatureCollection`] with each *element* of the
+//! [`GeometryCollection`](Value::GeometryCollection) converted to it's own [`Feature`], potentially
```
`its` rather than `it's`

urschrei commented on src/lib.rs:
```diff
@@ -14,147 +14,140 @@
+//!
+//! let mut properties = serde_json::Map::new();
+//! let key = "name".to_string();
+//! let value = "Fireston Grill".to_string();
```
Fireston**e** Grill. I have no idea where these places came from.

urschrei commented on src/lib.rs:
```diff
@@ -185,8 +178,8 @@
-//!             // GeometryCollections contain other Geometry types, and can nest
-//!             // we deal with this by recursively processing each geometry
+//!             // !!! GeometryCollections contain other Geometry types, and can
+//!             // nest we deal with this by recursively processing each geometry
```
Maybe some sort of separator after `geometry`: `:` or `–`?

urschrei reviewed (approved):
Just some very minor suggestions, so take or leave them as you see fit. The only other issue relates to `quick_collection` which IIRC can be a bit less efficient than manually processing geometries – not sure whether we want to call that out.

In any case, this is a **vast** improvement in clarity. Thank you!

michaelkirk commented:
> Just some very minor suggestions,

Thank you very much for the close proofread! I incorporated all of your suggestions and squashed.

bors r=lnicola, urschrei
